packages/jest-jasmine2/src/jasmine/Suite.js

```diff
@@ -181,6 +181,25 @@ Suite.prototype.addExpectationResult = function() {
   }
 };
 
+function convertDescriptorToString(descriptor) {
+  if (typeof descriptor === 'string' || descriptor === undefined) {
+    return descriptor;
+  }
+
+  if (descriptor.name !== undefined) {
+    return descriptor.name;
+  }
+
+  const stringified = descriptor.toString();
+  const typeDescriptorMatch = stringified.match(/class|function/);
```

Comment:
If something silly, such as a number or `{ toString() { return "nope"; }` is passed here, it'll crash. How would you like this function to deal with those error handling cases?

Reply:
Can you just make the function throw in that case and ask people to provide a string, class or function instead?